tests/unit/test_config_provider.py

```diff
@@ -942,3 +943,187 @@ def test_resolve_auto_mode_imds_region_provider_connect_timeout(self):
+            expected_value='global-from-config',
+        )
+
+    def test_can_provide_service_config_var(self):
+        full_config_map = {
+            'profiles': {'default': {'services': 'my-services'}},
+            'services': {
+                'my-services': {
+                    'batch': {'endpoint_url': 'global-from-config'}
+                }
+            },
+        }
+
+        scoped_config_map = full_config_map['profiles']['default']
+
+        self.assert_does_provide(
+            service='batch',
+            environ_map={},
+            full_config_map=full_config_map,
+            scoped_config_map=scoped_config_map,
+            expected_value='global-from-config',
+        )
+
+    def test_can_provide_service_config_var_over_global(self):
+        full_config_map = {
+            'profiles': {
+                'default': {
+                    'services': 'my-services',
+                    'endpoint_url': 'global-from-config',
+                }
+            },
+            'services': {
+                'my-services': {'batch': {'endpoint_url': "batch-from-config"}}
+            },
+        }
+
+        scoped_config_map = full_config_map['profiles']['default']
+        self.assert_does_provide(
+            service='batch',
+            environ_map={},
+            full_config_map=full_config_map,
+            scoped_config_map=scoped_config_map,
+            expected_value='batch-from-config',
+        )
+
+    def test_can_provide_service_config_var_over_global_diff_service(self):
+        full_config_map = {
+            'profiles': {
+                'default': {
+                    'services': 'my-services',
+                    'endpoint_url': 'global-from-config',
+                }
+            },
+            'services': {
+                'my-services': {'batch': {'endpoint_url': "batch-from-config"}}
+            },
+        }
+
+        scoped_config_map = full_config_map['profiles']['default']
+
+        self.assert_does_provide(
+            service='s3',
+            environ_map={},
+            full_config_map=full_config_map,
+            scoped_config_map=scoped_config_map,
+            expected_value='global-from-config',
+        )
+
+    def test_is_deepcopyable(self):
+        env = {'AWS_ENDPOINT_URL_BATCH': 'https://endpoint-override'}
+        provider = ConfiguredEndpointProvider(
+            full_config={}, scoped_config={}, client_name='batch', environ=env
+        )
+
+        provider_deepcopy = copy.deepcopy(provider)
+        assert provider is not provider_deepcopy
+        assert provider.provide() == 'https://endpoint-override'
+        assert provider_deepcopy.provide() == 'https://endpoint-override'
+
+        env['AWS_ENDPOINT_URL_BATCH'] = 'https://another-new-endpoint-override'
+        assert provider.provide() == 'https://another-new-endpoint-override'
+        assert provider_deepcopy.provide() == 'https://endpoint-override'
+
+
+@pytest.mark.parametrize(
```

Comment:
Let's shelf this test for the functional test PR. In general, unit tests should not be dependent on service models. That is reserved more for functional tests. I think when we do port this to be a functional test we would also use the client interface so that we can have a close to an end to end guarantee as we can get.

We should also make sure to include any service aliases in this test (e.g. `sagemaker.runtime`)